rocAL/source/loaders/image/image_loader.cpp

```diff
@@ -38,6 +38,10 @@ ImageLoader::ImageLoader(void *dev_resources) : _circ_buff(dev_resources),
     _is_initialized = false;
     _remaining_image_count = 0;
     _device_id = 0;
+#if ENABLE_HIP
+    DeviceResourcesHip *hipres = static_cast<DeviceResourcesHip *>(dev_resources);
```

Comment:
Is there any other way we can use existing HIP stream from circular buffer or master_graph? We already create 2 HIP streams in rocAL before this and RPP creates one for HIP processing. Will creating multiple HIP streams reduce HIP performance?

Reply:
We are not creating new stream here, we are using the existing stream [here ](https://github.com/fiona-gladwin/rocAL/blob/af0a0be4faf67a74339020769b38470c7ff1359f/rocAL/source/loaders/image/image_loader.cpp#L148)that is used in circular buffer